app/src/main/java/helium314/keyboard/latin/inputlogic/InputLogic.java

```diff
@@ -617,6 +617,11 @@ private void handleConsumedEvent(final Event event, final InputTransaction input
             inputTransaction.setDidAffectContents();
         }
         if (mWordComposer.isComposingWord()) {
+            // Check if we need to insert automatic space before starting to compose (e.g., after suggestion pickup)
+            if (SpaceState.PHANTOM == inputTransaction.getMSpaceState()) {
```

Comment:
Could you make sure this only happening when the Khipro combiner is used? To check you could expose `mCombiningSpec` in the `WordComposer`, e.g. add a `WordComposer.getCombiningSpec()` method.